### PR TITLE
Refactor input options

### PR DESCRIPTION
**Pull request by tgauth**
# PR Summary
- address #226 

## PR Context
- move `--input-file`/`--input` from global commands to relevant subcommands

### CODE REVIEW

File: dsc/src/args.rs [hunk 1]
```diff
@@ -77,15 +73,40 @@ pub enum SubCommand {
 #[derive(Debug, PartialEq, Eq, Subcommand)]
 pub enum ConfigSubCommand {
     #[clap(name = "get", about = "Retrieve the current configuration")]
-    Get,
+    Get {
+        #[clap(short = 'd', long, help = "The input to pass to the configuration or resource", conflicts_with = "path")]
```
Comment (SteveL-MSFT) [inline dsc/src/args.rs:77]:
```suggestion
        #[clap(short = 'd', long, help = "The document to pass to the configuration or resource", conflicts_with = "path")]
```

Change everywhere

File: dsc/src/subcommand.rs [hunk 2]
```diff
@@ -411,20 +394,20 @@ pub fn resource(subcommand: &ResourceSubCommand, format: &Option<OutputFormat>,
                 table.print();
             }
         },
-        ResourceSubCommand::Get { resource, input, all } => {
+        ResourceSubCommand::Get { resource, input, path, all } => {
```
Comment (SteveL-MSFT) [inline dsc/src/subcommand.rs:397]: Should we have a function that takes `input` (maybe needs to be renamed to `document` now?), `stdin`, and `path`, and just returns the string contents so that the `get`, `set`, etc... functions just take one `input` parameter instead of all of them?  You already have the `get_input()` below that can be used.

File: dsc/src/util.rs [hunk 3]
```diff
@@ -288,3 +289,61 @@ pub fn enable_tracing(trace_level: &TraceLevel, trace_format: &TraceFormat) {
...
+            return String::new();
+        },
+        _default => {
+            /* clap should handle these cases via conflicts_with so this should not get reached */
+            error!("Error: Invalid input");
+            exit(EXIT_INVALID_ARGS);
+        }
+    };
+
+    if value.is_empty() {
```
Comment (SteveL-MSFT) [inline dsc/src/util.rs:344]: Unclear to me when this value is empty?  Maybe add a comment.  Should also have test cases when any of the inputs is empty string.

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: dsc/src/util.rs (hunk 3 above)
Comment (tgauth) [reply dsc/src/util.rs:344]: I'm not completely sure either - I put it in because the existing dsc_args.tests.ps1 were failing on the CI but not locally within `get_inputs` - it seems like it was detecting `Some` for both `stdin` and either `document` or `path` but `stdin` was actually empty.

Comment (SteveL-MSFT) [reply dsc/src/util.rs:344]: I think you can technically pass an empty string via stdin: "" | dsc, although not entirely sure

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: dsc/src/util.rs [hunk 4]
```diff
@@ -288,3 +289,71 @@ pub fn enable_tracing(trace_level: &TraceLevel, trace_format: &TraceFormat) {
...
+}
+
+pub fn get_input(input: &Option<String>, stdin: &Option<String>, path: &Option<String>) -> String {
+    let value = match (input, stdin, path) {
+        (Some(_), Some(_), None) | (None, Some(_), Some(_)) => {
+            error!("Error: Cannot specify both stdin and --input or --path");
+            exit(EXIT_INVALID_ARGS);
+        },
+        (Some(input), None, None) => {
+            info!("Reading input from command line parameter");
```
Comment (SteveL-MSFT) [inline dsc/src/util.rs:323]: `info!()` should still be end-user useful.  You should use `debug!()` here.

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

**Review by SteveL-MSFT — APPROVED** [review]
